topology/src/main/java/io/camunda/zeebe/topology/ClusterTopologyManagerImpl.java

```diff
@@ -276,6 +279,15 @@ private void onOperationApplied(
     if (error == null) {
       observer.applied();
       backoffRetry.reset();
+      if (persistedClusterTopology.getTopology().version()
+          != topologyOnWhichOperationIsApplied.version()) {
+        LOG.debug(
+            "Topology changed while applying operation {}. Expected topology is {}. Current topology is {}. Most likely the change operation was cancelled.",
+            operation,
+            topologyOnWhichOperationIsApplied,
+            persistedClusterTopology.getTopology());
+        return;
+      }
       updateLocalTopology(
           persistedClusterTopology.getTopology().advanceTopologyChange(localMemberId, transformer));
```

Comment:
:thought_balloon: Could we just make `updateLocalTopology` a CAS operation? So `onOperationApplied` just takes the expected current topology as input and passes it to `updateLocalTopology` which then compares it with the persisted one and bails out if it's different.
To me this would feel a bit more robust

Reply:
Not sure whether that is worth. `onOperationApplied` is the only place where this CAS is required. Other usages are directly updating the current topology.